### ISSUE TITLE
Could not resolve placeholder 'osiam.db.driver' in string value "${osiam.db.driver}"

### ISSUE THREAD

**Issue by harikrishna553** [osiam/osiam#219]
I am working in setting up osiam server, MySQL as backend. I am stuck with following error. Any help on this....

org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'databaseConfiguration': Injection of autowired dependencies failed; nested exception is org.springframework.beans.factory.BeanCreationException: Could not autowire field: private java.lang.String org.osiam.configuration.DatabaseConfiguration.driverClassName; nested exception is java.lang.IllegalArgumentException: Could not resolve placeholder 'osiam.db.driver' in string value "${osiam.db.driver}"

```
1. git clone https://github.com/osiam/osiam.git

2. Update osiam.home in context.xml

<Context>
    <Environment name="osiam.home" value="/Users/harikrishna_gurram/" type="java.lang.String"/>
</Context>

3. Update tomcat heap size: catalina.sh
export CATALINA_OPTS="-Xms512M -Xmx1024M"

4. Create database osiam and use named osiam
mysql> create database osiam;

Create a user and grant permissions
CREATE USER 'osiam'@'localhost' IDENTIFIED BY 'osiam';
GRANT ALL PRIVILEGES ON * . * TO 'osiam'@'localhost';

5. Updated ./src/main/resources/home/config/osiam.yamlfor MySQL configurations and commented H2 database configurations.

 vendor: mysql
    driver: com.mysql.jdbc.Driver
    url: jdbc:mysql://localhost/osiam
    username: osiam
    password: osiam

6. Generated war file.
/Users/harikrishna_gurram/Downloads/apache-maven-3.3.9/bin/mvn clean package

7. Deployed war file in tomcat.
```

**Comment by dacrome** [comment]
Hey @harikrishna553 

thanks for trying OSIAM.

I think you did not [initialize the OSIAM home directory](https://github.com/osiam/osiam/blob/master/docs/detailed-reference-installation.md#initialize-the-home-directory-from-the-command-line). In your case it should be:

`./osiam.war initHome --osiam.home=/Users/harikrishna_gurram`

This command extracts all configuration and asset files directly to the `/Users/harikrishna_gurram` directory. You may want to choose an own directory for the files instead of you home directory e.g. `/Users/harikrishna_gurram/osiam`

But actually there is no need and it's not recommended to change the `osiam.yml` within the sources before generating the war file, when you initialize the OSIAM home directory.
You do not even generate the war file, you can download the [latest snapshot from jfrog](https://oss.jfrog.org/artifactory/repo/org/osiam/osiam/3.0-SNAPSHOT).

So the steps should be:

```
1. Download the latest snapshot from jfrog: https://oss.jfrog.org/artifactory/repo/org/osiam/osiam/3.0-SNAPSHOT

2. Update osiam.home in context.xml

<Context>
    <Environment name="osiam.home" value="/Users/harikrishna_gurram/" type="java.lang.String"/>
</Context>

3. Update tomcat heap size: catalina.sh
export CATALINA_OPTS="-Xms512M -Xmx1024M"

4. Create database osiam and use named osiam
mysql> create database osiam;

Create a user and grant permissions
CREATE USER 'osiam'@'localhost' IDENTIFIED BY 'osiam';
GRANT ALL PRIVILEGES ON * . * TO 'osiam'@'localhost';

5. Run on command line: ./osiam.war initHome --osiam.home=/Users/harikrishna_gurram

6. Updated ./Users/harikrishna_gurram/config/osiam.yaml for MySQL configurations and commented (or remove) H2 database configurations.

 vendor: mysql
    driver: com.mysql.jdbc.Driver
    url: jdbc:mysql://localhost/osiam
    username: osiam
    password: osiam

7. Deployed war file in tomcat.
```

Hope this helps!

**Comment by harikrishna553** [comment]
Thanks, It helps....:)

**Comment by ravesingh1983** [comment]
Hi dacrome,

I am facing similar issues but not exactly the same. I have set oisam.home properly. 
Also i am running osiam inside the tomcat. What happened is my tomcat did not start properly but I saw in the tomcat logs that it tried to deploy osiam and i can see the particular line.

2018-08-06 05:23:26.882  INFO 10313 --- [ost-startStop-1] org.osiam.OsiamHome                      : osiam.home = /opt/test/osiam/oauth

But when i go to osiam.home, I donot see anything and because of which i am getting the below error.

Any help is appreciated

org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'databaseConfiguration': Injection of autowired dependencies failed; nested exception is org.springframework.beans.factory.BeanCreationException: Could not autowire field: private java.lang.String org.osiam.configuration.DatabaseConfiguration.driverClassName; nested exception is java.lang.IllegalArgumentException: Could not resolve placeholder 'osiam.db.driver' in string value "${osiam.db.driver}"
